libs/mpi/CMakeLists.txt

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) 2019 The STE||AR-Group
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+cmake_minimum_required(VERSION 3.3.2 FATAL_ERROR)
```

Comment:
This line should probably go. It is overriding the policies set in the top CMakeLists.txt file. In particular, policy CMP0074 which is useful when MPI is installed in a custom directory: the MPI_ROOT variable is ignored. Moreover, the version specified is less than the version of the top CMake file.